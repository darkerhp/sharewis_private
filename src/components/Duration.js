import React from 'react';
import ReactNative from 'react-native';
import moment from 'moment';
import momentDurationFormat from 'moment-duration-format'; // eslint-disable-line

const { PropTypes } = React;
const { View, StyleSheet, Text } = ReactNative;

const styles = StyleSheet.create({
  container: {},
  duration: {
    padding: 3,
    fontSize: 8,
    backgroundColor: '#F2F2F2',
    fontWeight: '600',
  },
});

const Duration = ({
  duration,
  format = 'mm:ss',
  containerStyleId = null,
  durationStyleId = null,
  prefixText = '',
}) =>
  <View style={[styles.container, containerStyleId]}>
    <Text style={[styles.duration, durationStyleId]}>
      {prefixText}{moment.duration(duration, 'seconds').format(format, { trim: false })}
    </Text>
  </View>;

Duration.propTypes = {
  duration: PropTypes.number.isRequired,
  format: PropTypes.string,
  containerStyleId: PropTypes.number,
  durationStyleId: PropTypes.number,
  prefixText: PropTypes.string,
};

export default Duration;
