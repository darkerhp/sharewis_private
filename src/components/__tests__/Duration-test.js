/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import Duration from '../Duration';

describe('Duration', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Duration estimatedTime={100} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
