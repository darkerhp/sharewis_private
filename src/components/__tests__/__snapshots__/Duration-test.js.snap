exports[`Duration renders correctly 1`] = `
<View
  style={null} />
`;
